common/djangoapps/student/tasks.py

```diff
@@ -48,3 +58,228 @@ def send_activation_email(self, subject, message, from_address, dest_addr):
             exc_info=True
         )
         raise Exception
+
+
+@receiver(ENROLL_STATUS_CHANGE)
+def update_sailthru(sender, event, user, mode, course_id, **kwargs):
+    """
+    Receives signal and calls a celery task to update the
+    enrollment track
+    Arguments:
+        user: current user
+        course_key: course key of a course
+    Returns:
+        None
+    """
+    update_course_enrollment.delay(user.email, course_id, mode)
+
+
+@task(bind=True)
+def update_course_enrollment(self, email, course_key, mode):
+    """Adds/updates Sailthru when a user adds to cart/purchases/upgrades a course
+         Args:
+            user: current user
+            course_key: course key of course
+        Returns:
+            None
+        """
+    course_url = build_course_url(course_key)
+    config = EmailMarketingConfiguration.current()
+
+    try:
+        sailthru_client = SailthruClient(config.sailthru_key, config.sailthru_secret)
+    except:
+        return
+
+    new_enroll = False
+    send_template = None
+    cost_in_cents = None
+    if mode == 'audit' or mode == 'honor':
+        # free enroll
+        new_enroll = True
+        send_template = config.sailthru_enroll_template
```

Comment:
I believe the code in ecom-worker https://github.com/edx/ecommerce-worker/blob/master/ecommerce_worker/sailthru/v1/tasks.py#L263 test out condition more than these. Why only test out `audit` or `honor`?

Reply:
Now, it make sense to remove this condition https://github.com/edx/ecommerce-worker/blob/master/ecommerce_worker/sailthru/v1/tasks.py#L267 as now, there is no need to check whether, mode is audit or honor